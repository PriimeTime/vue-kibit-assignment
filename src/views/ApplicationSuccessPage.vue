<script setup lang="ts">
import DefaultLayout from "@/layouts/DefaultLayout.vue";
import BaseCard from "@/components/BaseCard.vue";
import HeaderText from "@/components/HeaderText.vue";
import BaseButton from "@/components/BaseButton.vue";
import { useRouter } from "vue-router";

const router = useRouter();
</script>

<template>
  <DefaultLayout>
    <section class="w-full flex justify-center items-center mt-10">
      <BaseCard class="p-6 md:shadow">
        <div class="space-y-4">
          <HeaderText class="text-2xl font-bold text-gray-800 text-center">
            Your application has been submitted successfully!
          </HeaderText>
          <BaseButton @click="router.push('/')" class="w-full">
            Go back to home
          </BaseButton>
        </div>
      </BaseCard>
    </section>
  </DefaultLayout>
</template>
