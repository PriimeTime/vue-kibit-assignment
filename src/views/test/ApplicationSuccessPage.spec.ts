import { describe, it, expect, vi } from "vitest";
import { mount } from "@vue/test-utils";
import ApplicationSuccessPage from "@/views/ApplicationSuccessPage.vue";
import { createRouter, createWebHistory } from "vue-router";
import { createTestingPinia } from "@pinia/testing";

const mockRouter = createRouter({
  history: createWebHistory(),
  routes: [{ path: "/", component: { template: "<div>Home</div>" } }],
});

describe("ApplicationSuccessPage", () => {
  it("renders without crashing", () => {
    const wrapper = mount(ApplicationSuccessPage, {
      global: {
        plugins: [
          mockRouter,
          createTestingPinia({
            createSpy: vi.fn,
          }),
        ],
      },
    });
    expect(wrapper.exists()).toBe(true);
  });

  it("displays the success message", () => {
    const wrapper = mount(ApplicationSuccessPage, {
      global: {
        plugins: [
          mockRouter,
          createTestingPinia({
            createSpy: vi.fn,
          }),
        ],
      },
    });
    expect(wrapper.text()).toContain(
      "Your application has been submitted successfully!"
    );
  });

  it("navigates to the home page when button is clicked", async () => {
    const pushMock = vi.spyOn(mockRouter, "push");

    const wrapper = mount(ApplicationSuccessPage, {
      global: {
        plugins: [
          mockRouter,
          createTestingPinia({
            createSpy: vi.fn,
          }),
        ],
      },
    });

    const button = wrapper.find("button");
    await button.trigger("click");

    expect(pushMock).toHaveBeenCalledWith("/");
  });

  it("matches snapshot", () => {
    const wrapper = mount(ApplicationSuccessPage, {
      global: {
        plugins: [
          mockRouter,
          createTestingPinia({
            createSpy: vi.fn,
          }),
        ],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
