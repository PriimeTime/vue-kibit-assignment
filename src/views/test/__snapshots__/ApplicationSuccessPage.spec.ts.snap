// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ApplicationSuccessPage > matches snapshot 1`] = `
"<header class="flex h-16 bg-teal-900">
  <div class="flex items-center w-2/3 text-xl md:text-3xl whitespace-nowrap text-white p-2 hover:cursor-pointer"> Job Ad Plaform </div>
  <div class="flex justify-end w-1/3 p-2"><button class="p-2 w-20 lg:mx-2 rounded-md text-white text-xs md:text-xl border-teal-900 border-2 bg-teal-900 flex justify-center items-center whitespace-nowrap">Home</button>
    <!--v-if--><button class="p-2 w-20 lg:mx-2 rounded-md text-white text-xs md:text-xl border-teal-900 border-2 bg-teal-900 flex justify-center items-center whitespace-nowrap">Login</button>
  </div>
</header>
<main>
  <section class="w-full flex justify-center items-center mt-10">
    <div class="max-w-[40rem] rounded-xl md:shadow-xl p-10 w-full p-6 md:shadow">
      <div class="space-y-4">
        <div class="text-3xl font-bold text-2xl font-bold text-gray-800 text-center"> Your application has been submitted successfully! </div><button class="p-2 min-w-20 rounded-md text-white border-teal-900 border-2 bg-teal-900 hover:text-teal-900 hover:bg-white flex justify-center items-center whitespace-nowrap w-full"> Go back to home </button>
      </div>
    </div>
  </section>
</main>"
`;
